Build these templates in out/ to make it easier to install them. Otherwise they are installed in the buildroot which would make for a messy install file.

Index: happy-1.18.6/Setup.lhs
===================================================================
--- happy-1.18.6.orig/Setup.lhs	2010-11-17 17:00:24.000000000 +0530
+++ happy-1.18.6/Setup.lhs	2011-03-29 13:58:16.000000000 +0530
@@ -33,9 +33,9 @@
         runProgram ghcProgram (["-o", dst, "-E", "-cpp", "templates" </> src] ++ opts)
	runProgram perlProgram ["-i.bak", "-pe", crazy_perl_regexp, dst]

-  sequence_ ([ cpp_template "GenericTemplate.hs" dst opts | (dst,opts) <- templates ] ++
-             [ cpp_template "GLR_Base.hs"       dst opts | (dst,opts) <- glr_base_templates ] ++
-             [ cpp_template "GLR_Lib.hs"        dst opts | (dst,opts) <- glr_templates ])
+  sequence_ ([ cpp_template "GenericTemplate.hs" ("out/" ++ dst) opts | (dst,opts) <- templates ] ++
+             [ cpp_template "GLR_Base.hs"       ("out/" ++ dst) opts | (dst,opts) <- glr_base_templates ] ++
+             [ cpp_template "GLR_Lib.hs"        ("out/" ++ dst) opts | (dst,opts) <- glr_templates ])

 myPostClean _ _ _ _ = mapM_ (try . removeFile) all_template_files

